File: test/migrate.test.js
import test from "node:test";
import assert from "node:assert/strict";
import * as migrate from "../brain_tools/migrate.js";
import { CURRENT_SCHEMA_VERSION } from "../brain_tools/schema.js";

test("migrate sets current version", () => {
  const out = migrate.migrateNote({ date: "2026-05-31", summary: "x" });
  assert.equal(out.schema_version, CURRENT_SCHEMA_VERSION);
});

test("migrate adds missing keywords", () => {
  const out = migrate.migrateNote({ schema_version: 0, date: "2026-05-31", summary: "x" });
  assert.deepEqual(out.keywords, []);
});

test("migrate preserves existing fields", () => {
  const out = migrate.migrateNote({
    schema_version: 0,
    date: "2026-05-31",
    summary: "x",
    keywords: ["auth"],
    tags: ["java"],
  });
  assert.deepEqual(out.keywords, ["auth"]);
  assert.deepEqual(out.tags, ["java"]);
});

test("needsMigration detects old version", () => {
  assert.equal(migrate.needsMigration({ schema_version: 0 }), true);
  assert.equal(migrate.needsMigration({ schema_version: 1 }), false);
});

File: src/test/java/brain/MigrateTest.java
package brain;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertFalse;
import static org.junit.jupiter.api.Assertions.assertTrue;

import java.util.List;
import java.util.Map;
import org.junit.jupiter.api.Test;

class MigrateTest {
  @Test
  void migrateSetsCurrentVersion() {
    Map<String, Object> out = Migrate.migrateNote(Map.of("date", "2026-05-31", "summary", "x"));
    assertEquals(Schema.CURRENT_SCHEMA_VERSION, out.get("schema_version"));
  }

  @Test
  void migrateAddsMissingKeywords() {
    Map<String, Object> out =
        Migrate.migrateNote(Map.of("schema_version", 0, "date", "2026-05-31", "summary", "x"));
    assertEquals(List.of(), out.get("keywords"));
  }

  @Test
  void migratePreservesExistingFields() {
    Map<String, Object> out = Migrate.migrateNote(Map.of(
        "schema_version", 0, "date", "2026-05-31", "summary", "x",
        "keywords", List.of("auth"), "tags", List.of("java")));
    assertEquals(List.of("auth"), out.get("keywords"));
    assertEquals(List.of("java"), out.get("tags"));
  }

  @Test
  void needsMigrationDetectsOldVersion() {
    assertTrue(Migrate.needsMigration(Map.of("schema_version", 0)));
    assertFalse(Migrate.needsMigration(Map.of("schema_version", 1)));
  }
}
